script/tool/test/format_command_test.dart

```diff
@@ -85,12 +86,21 @@ void main() {
       'lib/src/b.dart',
       'lib/src/c.dart',
     ];
+    const String fileWithOptOut = 'lib/src/d.dart';
     final Directory pluginDir = createFakePlugin(
       'a_plugin',
       packagesDir,
-      extraFiles: files,
+      extraFiles: <String>[
+        ...files,
+        fileWithOptOut,
+      ],
     );
 
+    final p.Context posixContext = p.posix;
+    childFileWithSubcomponents(pluginDir, posixContext.split(fileWithOptOut))
+        .writeAsStringSync(
+            '// copyright bla bla\n// This file is hand-formatted.\ncode...');
```

Comment:
This is a distinct enough thing from the basic "does the formatter get called" that I think it would be much better to have a separate "skips files with the opt-out" test, so that a test failure would be more specific.